packages/docs/src/components/docsearch/doc-search.tsx

```diff
@@ -64,43 +64,45 @@ export const DocSearch = component$((props: DocSearchProps) => {
     snippetLength: 10,
   });
 
-  const searchButtonRef = useRef();
+  const searchButtonRef = useSignal();
 
   return (
     <div
       class="docsearch"
+      preventdefault:keyDown={!state.isOpen}
       window:onKeyDown$={(event) => {
-        function open() {
-          // We check that no other DocSearch modal is showing before opening
-          // another one.
-          if (!document.body.classList.contains('DocSearch--active')) {
-            state.isOpen = true;
+        window.addEventListener('keydown', (event) => {
```

Comment:
imo this line should be obsolete now since we already have a `keyDown` listener registered on L73 😃

Reply:
ty for pointing out, my bad tho it doesn't work again when using ctrl-k to close the doc search component :(

I think we should bind it to window only rather than the div

https://user-images.githubusercontent.com/70624701/222942166-55cd1194-0272-4276-9a1e-4276391a9228.mp4

Reply:
oh no 🤣  i mean we could try to debounce the update but this is a bit over the top. i am also not sure, why this fn needs so many triggers. you can open it atm using `cmd + k` or `/`. to close it you can hit `cmd + k` and `esc`.
just compared it with https://vercel.com/docs and it has the same behavior as on your video.
for me the important thing is, that you can open it with that shortcut. the goal is not to close it right away again but finding the page you're looking for 😄 so maybe we could remove `cmd + k` from the close cmd as well.

Reply:
yeah i agree with you, i think it would be better to just rm `cmd + k` binding 😄

Reply:
Hi @vedantnn71 sry for the late answer 🙈 
thanks for the replies 🙏  imo we should remove the `window.eventListener` as mentioned above and also the `event.preventDefault()` on line 90. and then we should be defenetely good to go 🚀

Reply:
ty @zanettin, rmed prevent default 😄